cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -20,18 +20,30 @@
 /**
  * Defines a cron-based schedule for running a program. 
  */
-public class Schedule {
+public abstract class Schedule {
+  /**
+   * Schedule Type.
+   */
+  public enum ScheduleType { TIME, DATA }
```

Comment:
Looking at the code, it seems that his only used internally in subclasses of Schedule. So it could be protected? I am trying to leak as little of the implementation to the API as possible.